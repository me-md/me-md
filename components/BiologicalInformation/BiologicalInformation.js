import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Button } from 'native-base';
import { LinearGradient } from 'expo-linear-gradient';
import { MaterialCommunityIcons } from '@expo/vector-icons';

export default class BiologicalInformation extends Component {
  render() {
    return (
      <View style={styles.biologicalInformationContainer}>
        <LinearGradient
          style={styles.gradientBackground}
          colors={['#004EFF', '#88CCF1']}
        >
          <View style={styles.biologicalInformationAnswer}>
            <Text style={styles.biologicalInformationQuestion}>
              What is your biological sex?
            </Text>
            <Button iconLeft style={styles.biologicalInformationButtons}>
              <MaterialCommunityIcons
                name='gender-female'
                size={64}
                color='black'
              />
              <Text style={styles.biologicalInformationButtonText}>Female</Text>
            </Button>
            <Button iconLeft style={styles.biologicalInformationButtons}>
              <MaterialCommunityIcons
                name='gender-male'
                size={64}
                color='black'
              />
              <Text style={styles.biologicalInformationButtonText}>Male</Text>
            </Button>
          </View>
        </LinearGradient>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  biologicalInformationAnswer: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'space-around',
    paddingTop: 75
  },
  biologicalInformationButtons: {
    alignSelf: 'center',
    backgroundColor: 'rgba(255, 255, 255, 0.45)',
    borderRadius: 50,
    height: 150,
    margin: 25,
    padding: 20,
    width: 275
  },
  biologicalInformationContainer: {
    flex: 1
  },
  biologicalInformationQuestion: {
    alignItems: 'center',
    color: '#fff',
    flex: 1,
    fontSize: 48,
    height: 30,
    justifyContent: 'center'
  },
  biologicalInformationButtonText: {
    fontSize: 36
  },
  gradientBackground: {
    flex: 1
  }
});
