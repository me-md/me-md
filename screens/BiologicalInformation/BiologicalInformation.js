import React, { useState } from 'react';
import { Dimensions, StyleSheet, Text, View } from 'react-native';
import { Button } from 'native-base';
import { LinearGradient } from 'expo-linear-gradient';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { Entypo } from '@expo/vector-icons';

const height = Dimensions.get('window').height;

export default function BiologicalInformation({ navigation }) {
  const [sex, setSex] = useState('');

  return (
    <View style={styles.container}>
      <LinearGradient
        style={styles.gradientBackground}
        colors={['#004EFF', '#88CCF1']}
      >
        <View style={styles.biologicalInformationContainer}>
          <Entypo
            name='chevron-thin-up'
            size={36}
            color='white'
            onPress={() => navigation.goBack()}
          />
          <View style={styles.buttonContainer}>
            <Text style={styles.question}>What is your biological sex?</Text>
            <Button
              rounded
              style={
                sex === 'female'
                  ? styles.pressed
                  : styles.biologicalInformationButtons
              }
              onPress={() => (sex === 'female' ? setSex('') : setSex('female'))}
            >
              <MaterialCommunityIcons
                name='gender-female'
                size={64}
                color='white'
              />
              <Text style={styles.biologicalInformationButtonText}>Female</Text>
            </Button>
            <Button
              rounded
              style={
                sex === 'male'
                  ? styles.pressed
                  : styles.biologicalInformationButtons
              }
              onPress={() => (sex === 'male' ? setSex('') : setSex('male'))}
            >
              <MaterialCommunityIcons
                name='gender-male'
                size={64}
                color='white'
              />
              <Text style={styles.biologicalInformationButtonText}>Male</Text>
            </Button>
          </View>
          {sex !== '' ?
            <Entypo
              name='chevron-thin-down'
              size={36}
              color='white'
              onPress={() => {
                navigation.push('SelectAge', { sex })
              }}
            /> : <Entypo
              name='chevron-thin-down'
              size={36}
              style={{ opacity: 0 }}
              color='grey'
            />}
        </View>
      </LinearGradient>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  gradientBackground: {
    flex: 1
  },
  biologicalInformationContainer: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'space-between',
    marginBottom: height * 0.06,
    marginTop: height * 0.06,
  },
  buttonContainer: {
    justifyContent: 'space-between'
  },
  question: {
    color: '#FFFFFF',
    fontSize: 48,
    marginBottom: height * 0.07
  },
  biologicalInformationButtons: {
    backgroundColor: 'rgba(255, 255, 255, 0.1)',
    height: 100,
    margin: 25,
    padding: 20,
    width: 225
  },
  pressed: {
    backgroundColor: 'rgba(255, 255, 255, 0.5)',
    height: 100,
    margin: 25,
    padding: 20,
    width: 225
  },
  biologicalInformationButtonText: {
    color: '#FFFFFF',
    fontSize: 36
  }
});
