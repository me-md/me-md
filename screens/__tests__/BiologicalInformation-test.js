import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import BiologicalInformation from '../../screens/BiologicalInformation/BiologicalInformation';

jest.mock('expo', () => ({
  AppLoading: 'AppLoading',
}));

describe('BiologicalInformation', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it(`should match a snapshot`, () => {
    const tree = renderer.create(<BiologicalInformation />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
