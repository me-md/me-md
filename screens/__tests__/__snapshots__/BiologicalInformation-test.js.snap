// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BiologicalInformation should match a snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4278210303,
          4287155441,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-between",
          "marginBottom": 80.03999999999999,
          "marginTop": 80.03999999999999,
        }
      }
    >
      <Text />
      <View
        style={
          Object {
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 48,
              "marginBottom": 93.38000000000001,
            }
          }
        >
          What is your biological sex?
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 255, 255, 0.1)",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 57,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 100,
              "justifyContent": "space-between",
              "margin": 25,
              "opacity": 1,
              "padding": 20,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
              "width": 225,
            }
          }
        >
          <Text />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 36,
              }
            }
          >
            Female
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 255, 255, 0.1)",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 57,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 100,
              "justifyContent": "space-between",
              "margin": 25,
              "opacity": 1,
              "padding": 20,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
              "width": 225,
            }
          }
        >
          <Text />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 36,
              }
            }
          >
            Male
          </Text>
        </View>
      </View>
      <Text />
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
